pkg/agent/baker.go

```diff
@@ -964,6 +964,12 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 		"GetPrivateEgressProxyAddress": func() string {
 			return config.ContainerService.Properties.SecurityProfile.GetProxyAddress()
 		},
+		"GetBootstrapProfileContainerRegistryServer": func() string {
```

Comment:
let's add a few new UTs to baker_test.go to ensure that `/etc/containerd/certs.d/mcr.microsoft.com/hosts.toml` is getting generated as we expect

Reply:
Thanks! Added the UT.

`/etc/containerd/certs.d/mcr.microsoft.com/hosts.toml` is generated as expected
https://github.com/Azure/AgentBaker/pull/4813/files#diff-a6f4d8f0504efa4d2a9814560ab246a790447d5cab922f79ae9f8ed643154e10R316